ui/src/containers/BeamlineSetupContainer.jsx

```diff
@@ -39,6 +39,25 @@ function BeamlineSetupContainer(props) {
     setAttribute(name, value);
   }
 
+  function handleSendCommand(targetState) {
+    switch (targetState) {
+      case 'READY': {
+        sendCommand('PowerOn');
+        break;
+      }
```

Comment:
I am Wondering , you send PowerOn when State is READY ?
I would have expert READY mean the Device is powered on , .

But good to know if it's the case

Reply:
No, I send PowerOn command when I want to _switch_ to READY state. The switch case is on target state.

In practice I send PowerOn when in DISABLED state.

Reply:
Ahh in that case you could  just change line 113 of the  InOutSwitch component to :
onClick={() => handleSwitch(openValue)}

That would be it .

Reply:
Ah, yeah that would be better.

I'll test it, next week when I'm back from my vacation :)

Reply:
After looking a bit closer at the InOutSwitch code, I think the right way to fix this is to treat 'DISABLED' state the same way as 'CLOSED' state.